src/Microsoft.ML.TimeSeries/SrCnnEntireAnomalyDetector.cs

```diff
@@ -127,29 +184,42 @@ protected override DataViewType GetColumnTypeCore(int iinfo)
             }
         }
 
-        public SrCnnEntireAnomalyDetector(IHostEnvironment env, IDataView input, string inputColumnName, string outputColumnName, double threshold, int batchSize, double sensitivity, SrCnnDetectMode detectMode)
+        public SrCnnEntireAnomalyDetector(IHostEnvironment env, IDataView input, string outputColumnName, string inputColumnName, SrCnnEntireAnomalyDetectorOptions options)
             : base(env, nameof(SrCnnEntireAnomalyDetector), input)
         {
+            Host.CheckValue(outputColumnName, nameof(outputColumnName));
+
             Host.CheckValue(inputColumnName, nameof(inputColumnName));
             _inputColumnName = inputColumnName;
 
-            Host.CheckUserArg(batchSize == -1 || batchSize >= MinBatchSize, nameof(batchSize), "BatchSize must be -1 or no less than 12.");
-            _batchSize = batchSize;
+            Host.CheckValue(options, nameof(options));
+            CheckOptionArguments(options);
 
-            Host.CheckUserArg(threshold >= 0 && threshold <= 1, nameof(threshold), "Must be in [0,1].");
-            Host.CheckUserArg(detectMode == SrCnnDetectMode.AnomalyOnly
-                || detectMode == SrCnnDetectMode.AnomalyAndExpectedValue
-                || detectMode == SrCnnDetectMode.AnomalyAndMargin, nameof(detectMode), "Invalid detectMode");
-
-            Host.CheckUserArg(sensitivity >= 0 && sensitivity <= 100, nameof(sensitivity), "Must be in [0,100].");
-            _outputLength = _outputLengthArray[(int)detectMode];
-            _threshold = threshold;
-            _sensitivity = sensitivity;
-            _detectMode = detectMode;
+            _options = options;
+            _outputLength = _outputLengthArray[(int)options.DetectMode];
 
             _bindings = new Bindings(input.Schema, inputColumnName, outputColumnName, new VectorDataViewType(NumberDataViewType.Double, _outputLength));
         }
 
+        private void CheckOptionArguments(SrCnnEntireAnomalyDetectorOptions options)
+        {
+            Host.CheckUserArg(options.Period >= 0, nameof(options.Period), "Must be integer equal or greater than 0.");
```

Comment:
Suggestion: Must be **an** integer equal **to** or greater than 0.

Reply:
Updated, thanks.

---
In reply to: [442532765](https://github.com/dotnet/machinelearning/pull/5202#discussion_r442532765) [](ancestors = 442532765)